docs/en/Unit-Testing.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: global
+title: Unit Testing
+nickname: Unit Testing
+group: Resources
+---
+
+* Table of Contents
+{:toc}
+
+# Unit testing goals
+1. Unit tests act as examples of how to use the code under test.
+2. Unit tests detect when an object breaks it's specification.
+3. Unit tests *don't* break when an object is refactored but still meets the same specification.
+
+# How to write a unit test
+
+0. If creating an instance of the class takes some work, create `@Before` and `@After` methods which handle setup and cleanup. Only do general setup which will apply to every test. Test-specific setup should be done locally in the tests that need it.
+
+	    @Before
+	    public void before() throws Exception {
+	      Journal blockJournal = new ReadWriteJournal(mTestFolder.newFolder().getAbsolutePath());
+	      mClock = new TestClock();
+	      mExecutorService =
+	          Executors.newFixedThreadPool(2, ThreadFactoryUtils.build("TestBlockMaster-%d", true));
+	      mMaster = new BlockMaster(blockJournal, mClock, mExecutorService);
+	      mMaster.start(true);
+	    }
+
+	    @After
+	    public void after() throws Exception {
+	      mMaster.stop();
+	    }
+
+1. Decide on an element of functionality to test. The functionality you decide to test should be part of the public API and should not care about implementation details.
+2. Give your test a name that describes what it’s testing. The functionality being tested should be simple enough to describe this way:
```

Comment:
Added some examples. The guideline is that the test should be named after the functionality that it's testing.